src/main/scala/com/fulcrumgenomics/bam/CallOverlappingConsensusBases.scala

```diff
@@ -0,0 +1,182 @@
+  private object ThreadData {
+    def apply(): ThreadData = ThreadData(
+      caller         = new OverlappingBasesConsensusCaller(onlyMaskDisagreements=onlyMaskDisagreements, maxQualOnAgreement=maxQualOnAgreement),
+      templateMetric = CallOverlappingConsensusBasesMetric(tpe=CountType.Templates),
+      basesMetric    = CallOverlappingConsensusBasesMetric(tpe=CountType.Bases)
+    )
+  }
+
+  override def execute(): Unit = {
+    val source           = SamSource(input)
+    val writer           = SamWriter(output, source.header)
+    val progress         = new ProgressLogger(logger)
+    val templateIterator = Bams.templateIterator(source)
+    val threadData       = new IterableThreadLocal(() => ThreadData())
+
+    ParIterator(templateIterator, threads=threads)
+      .map { template =>
+        val threadDatum = threadData.get()
+        threadDatum.synchronized {
+          // update metrics
+          threadDatum.templateMetric.total += 1
+          threadDatum.basesMetric.total += template.primaryReads.map(_.length).sum
+          // corrects
+          val stats = threadDatum.caller.call(template)
+          val overlappingBases = stats.r1OverlappingBases + stats.r2OverlappingBases
+          val correctedBases   = stats.r1CorrectedBases + stats.r2CorrectedBases
+          if (overlappingBases > 0) {
+            threadDatum.templateMetric.overlapping += 1
+            threadDatum.basesMetric.overlapping += overlappingBases
+            if (correctedBases > 0) {
+              threadDatum.templateMetric.corrected += 1
+              threadDatum.basesMetric.corrected += correctedBases
+            }
+          }
+        }
+        template.allReads.foreach(progress.record)
+        template
+      }.toAsync(ParIterator.DefaultChunkSize * 8).foreach { template =>
+      writer ++= template.allReads
+    }
+    progress.logLast()
+    source.safelyClose()
+    writer.close()
+
+    val templatesMetric = CallOverlappingConsensusBasesMetric(tpe=CountType.Templates)
+    val basesMetric     = CallOverlappingConsensusBasesMetric(tpe=CountType.Bases)
+    threadData.foreach { datum =>
+      templatesMetric += datum.templateMetric
+      basesMetric     += datum.basesMetric
+    }
+
+    Metric.write(metrics, templatesMetric, basesMetric)
+  }
+
+}
+
+/** Collects the the number of reads or bases that were examined, had overlap, and were corrected as part of
+  * the [[CallOverlappingConsensusBases]] tool.
+  *
+  * @param tpe template if the counts are per template, bases if counts are in units of bases.
+  * @param total the total number of templates (bases) examined
+  * @param overlapping the total number of templates (bases) that were overlapping
+  * @param corrected the total number of templates (bases) that were corrected.
+  */
+case class CallOverlappingConsensusBasesMetric
+(
+  tpe: CountType,
```

Comment:
I prefer to use `kind` here since it's not reserved and looks better than `tpe` in tsv files.